lib/elixir/lib/kernel.ex

```diff
@@ -2968,6 +2975,23 @@ defmodule Kernel do
 
       when x >= 1 and x <= 3
 
+  ### Special AST Considerations
+
+  `not in` is rewritten in the Elixir compiler, so `:left not in :right` appears
```

Comment:
I think we should use valid construction here, for example,  `1 not in [2, 3]`, version with atoms feels confusing.